google/web-stories-wp
# Allow re-triggering adaptive text style

## Summary
This PR implements allowing re-triggering adaptive text colors for a single text element.


## Relevant Technical Choices



## To-do



## User-facing changes
![adaptive](https://user-images.githubusercontent.com/3294597/137114946-4bb5b382-22e2-4b90-96a9-c5db7ddfabe5.gif)

<img width="470" alt="Screenshot 2021-10-13 at 13 19 12" src="https://user-images.githubusercontent.com/3294597/137115016-a3c405ce-3271-44b8-86c5-38273daa997f.png">



## Testing Instructions





This PR can be tested by following these steps:

1. Assign some image or dark color background to the page that would have very low contrast with default text color (black)
2. Now add a text element
3. Click on the adaptive text colors button in the design panel, verify the text color is changed as expected (see the screenshot/gif below for detecting the button)
4. Verify the same using the text element's quick action menu
5. Verify that the button is disabled for multi-selection of text elements.


### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
- Adds new tracking for using `auto_style_text` from the quick action menu
- Adds tracking `trackEvent('auto_style_text');` for using the button from the design panel.


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #9206
